sdk/core/System.ClientModel/src/Convenience/AsyncEnumerableResultOfT.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace System.ClientModel;
+
+#pragma warning disable CS1591 // public XML comments
+public abstract class AsyncEnumerableResult<T> : ClientResult, IAsyncEnumerable<T>, IDisposable
+    where T : IPersistableModel<T>
```

Comment:
Another question is whether or not we want to add this type constraint.  The benefit is that it could simplify implementations by allowing them to use ModelReaderWriter to deserialize values coming across the wire, rather than requiring them to use other ways to pass in bespoke deserialization routines.  A drawback is that it limits the use of the type to models that implement the interface -- if we wanted to enable scenarios like pages of primitives, that could be a problem.

Reply:
Since we did not constrain ```ClientResult<T>```, maybe we should keep not constraining. Though in retrospect, maybe we should have.

Reply:
The libraries haven't GA'ed yet, so we could still constrain it if we feel strongly about it.

Reply:
Ah, but not if we want Azure.Core Pageable\<T\> to be able to inherit from it.  Alas!  I will remove it.